tests/api.rs

```diff
@@ -0,0 +1,111 @@
+mod test_utils;
```

Comment:
It seems weird to move all these tests into the integration forlder when they aren't integration tests for the rpc

Reply:
but imo they're using the katana client, and behaving in "real" simulated situations, but true it's not integration test of the RPC itself

the reason behind it is that there is a library inside called test_utils, only compiled in the test integration folder, so if I move the tests outside of it they can't import it

Reply:
it's because the `test` config is only set for the crate that is being compiled. when you are integration tests, they are compiled as a separate crate, which imports the crate `test_utils` as a library, hence not having the `test` config set. I would personally keep the unit like tests in their respective folders.